src/EventSubscriber/RequestBodySubscriber.php

```diff
@@ -22,17 +26,29 @@ public static function getSubscribedEvents(): array
         ];
     }
 
-    public function onKernelController(FilterControllerEvent $event): void
+    public function onKernelController(ControllerEvent $event): void
     {
         $request = $event->getRequest();
         if ($request->getContentType() !== 'json' || !$request->getContent()) {
             return;
         }
 
-        $data = \json_decode($request->getContent(), true);
-        if (\json_last_error() !== JSON_ERROR_NONE) {
-            throw new BadRequestHttpException('Invalid json body: ' . json_last_error_msg());
+        try {
+            $data = \json_decode($request->getContent(), true, 512, JSON_THROW_ON_ERROR);
+        } catch (\JsonException $e) {
+            $sError = $e->getMessage();
+            Logger::info("Invalid json body: $sError", LogContext::SSK_BUNDLE(), [
+                "requestContent" => $request->getContent()
+            ]);
+
+            $response = new JsonResponse([
+                "s_message" => "Invalid request payload",
+                "error" => ["json decoding error" => [$sError]]
+            ], JsonResponse::HTTP_BAD_REQUEST);
+
+            $response->send();
         }
+
         $request->request->replace(\is_array($data) ? $data : []);
```

Comment:
```suggestion
        
```

Reply:
ah ok You want to move it inside catch in next comment

Reply:
Yep. Sorry, I was not clear.